service/src/main/java/bio/terra/cbas/controllers/util/AsyncExceptionHandler.java

```diff
@@ -1,21 +1,104 @@
 package bio.terra.cbas.controllers.util;
 
+import static bio.terra.cbas.models.CbasRunStatus.QUEUED;
+
+import bio.terra.cbas.common.DateUtils;
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.model.RunSetRequest;
+import bio.terra.cbas.models.CbasRunSetStatus;
+import bio.terra.cbas.models.CbasRunStatus;
+import bio.terra.cbas.models.Run;
+import bio.terra.cbas.models.RunSet;
 import java.lang.reflect.Method;
+import java.time.OffsetDateTime;
+import java.util.List;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.aop.interceptor.AsyncUncaughtExceptionHandler;
 
 public class AsyncExceptionHandler implements AsyncUncaughtExceptionHandler {
+  private final RunDao runDao;
+  private final RunSetDao runSetDao;
 
   private final Logger logger = LoggerFactory.getLogger(AsyncExceptionHandler.class);
 
+  private static final String STANDARD_LOG_MESSAGE =
+      "Exception thrown in Thread '%s' while executing method '%s'. Error message: %s";
+
+  public AsyncExceptionHandler(RunDao runDao, RunSetDao runSetDao) {
+    this.runDao = runDao;
+    this.runSetDao = runSetDao;
+  }
+
   @Override
   public void handleUncaughtException(Throwable ex, Method method, Object... params) {
-    // handle exception
-    logger.error(
-        "Exception thrown in Thread '{}' while executing method '{}'. Error message: {}",
-        Thread.currentThread().getName(),
-        method.getName(),
-        ex.getMessage());
+    String methodName = method.getName();
+    if (methodName.equals("triggerWorkflowSubmission")) {
+      handleExceptionFromAsyncSubmission(ex, methodName, params);
+    } else {
+      String logMsg =
+          STANDARD_LOG_MESSAGE.formatted(
+              Thread.currentThread().getName(), method.getName(), ex.getMessage());
+      logger.error(logMsg);
+    }
+  }
+
+  public void handleExceptionFromAsyncSubmission(
+      Throwable ex, String methodName, Object... params) {
+    RunSet runSet = null;
+    RunSetRequest runSetRequest = null;
+    var logMsg =
+        STANDARD_LOG_MESSAGE.formatted(
+            Thread.currentThread().getName(), methodName, ex.getMessage());
+
+    // extract method request parameters
+    for (Object o : params) {
+      if (o instanceof RunSetRequest) {
+        runSetRequest = (RunSetRequest) o;
+      }
+
+      if (o instanceof RunSet) {
+        runSet = (RunSet) o;
```

Comment:
Since I couldn't reply to your top-level comment in a thread, I'll reply here...

Yes I think you want to do something like 
```java
if (o instanceof RunSet runSet) {
  // ... use 'runSet' as a variable
}
```

It looks like maybe you're running into problems because you're then trying to say something like "if there were no runSets found, throw a different exception" below. You could have a flag variable like `boolean runSetFound` and set that to true if one of the params matched?

But presumably the params are always in the same order, so you should be able to predict which is the right one to key into? Or is there like a `paramName` map that also gets passed in somehow?

Reply:
Assuming the params are in order, I updated the code [this way](https://github.com/DataBiosphere/cbas/pull/281/files#diff-a44e5a98fe9164633883e906505b2cb3c7db64d37861c83d9eaeff45162f71a2R53-R83) and SonarCloud seems to be happy.